pkg/segment/aggregations/segaggs_test.go

```diff
@@ -1546,3 +1546,223 @@ func Test_getColumnsToKeepAndRemove(t *testing.T) {
 		})
 	}
 }
+
+func Test_performArithmeticOperation_Addition(t *testing.T) {
+	// Test addition with numbers
+	leftNum := 5
+	rightNum := 3
+	expectedNumResult := float64(leftNum + rightNum)
+	numResult, err := performArithmeticOperation(leftNum, rightNum, utils.Add)
+	if err != nil {
+		t.Errorf("Failed to add numbers: %v", err)
+	}
+	numResult = numResult.(float64)
+	if numResult != expectedNumResult {
+		t.Errorf("Addition of numbers incorrect, got: %v, want: %v", numResult, expectedNumResult)
```

Comment:
- Use `assert.Equal(t, expected, actual, "error message")` instead of `t.Errorf()`

- Do the same for all others.